docs/en/Community-Articles/2024-04-19-using-blob-storage-with-abp/post.md

```diff
@@ -0,0 +1,79 @@
+
+- #### Step 2: Create ProfileAppService
+
+Create the ProfileAppService class and derive it from the ApplicationService class. This class will perform the necessary operations to store and retrieve profile pictures.
+
+   ![Step 2](./images/step2.png)
+
+- #### Step 3: Blob Container Connection
+
+In the constructor method, inject the dependency into the IBlobContainer interface. This interface is used to interact with the Blob Container.
+
+   ![Step 3](./images/step3.png)
+
+- #### Step 4: Save Profile Picture
+
+The SaveProfilePictureAsync method is used to store the user's profile picture. A unique name is generated based on the user's credentials and the profile picture byte array with this name is saved in the Blob Container.
+
+   ![Step 4](./images/step4.png)
+
+- #### Step 5: Getting Profile Picture
+
+The GetProfilePictureAsync method is used to get the user's profile picture. A profile picture byte array is retrieved from the Blob Container with a specified name based on the user's credential.
+
+   ![Step 5](./images/step5.png)
+
+
+Finally, add controls in the user interface that will allow users to upload and view their profile pictures. These controls will perform the operations by calling the corresponding methods in the ProfileAppService class.
+
+These steps cover the basic steps to store user profile pictures with Blob Storage using the ABP Framework. [Check out the documentation for more information.](https://docs.abp.io/en/abp/latest/Blob-Storing)
+
+
+## What are the advantages/disadvantages of keeping in Blob Storage database? 
+
+#### Advantages:
+
+- Data Integrity and Relational Model: To ensure data integrity and preserve the relational model, it is important to store blob data in the database. This approach preserves the relationships between data and maintains the structural integrity of the database. 
+
+- A Single Storage Location: Storing blob data in the database allows you to collect all data in a single storage location. This simplifies the management of data and increases data integrity. 
+
+- Advanced Security Controls: Database systems often offer advanced security controls. Storing blob data in a database allows you to take advantage of these security features and ensures that data is accessed by authorized users. 
+
+#### Disadvantages: 
+
+- Performance Issues: Storing blob data in a database can negatively impact database performance. Oversized blob data can slow down query processing and reduce database performance. 
+
+- Storage Space Issue: Storing blob data in the database can increase the size of the database and require more storage space. This can increase storage costs and complicate infrastructure requirements. 
+
+- Backup and Recovery Challenges: Storing blob data in a database can make backup and recovery difficult. The large size of blob data can make backup and recovery time-consuming and data recovery difficult. 
+
+
+## Information about Blob Storage entagrations  
```

Comment:
It's a [good practise to having a paragraph](https://developers.google.com/tech-writing/two/large-docs) after the title to pre-explain the section:

```suggestion
## Other Blob Storage Providers

ABP Framework provides some other [pre-built storage providers](https://docs.abp.io/en/abp/latest/Blob-Storing#blob-storage-providers) besides the [Database Provider](https://docs.abp.io/en/abp/latest/Blob-Storing-Database) shown in this article:
```